src/Microsoft.AspNetCore.Razor.Runtime/Resources.resx

```diff
@@ -195,4 +195,16 @@
   <data name="ArgumentMustBeAnInstanceOf" xml:space="preserve">
     <value>Argument must be an instance of '{0}'.</value>
   </data>
+  <data name="TagHelperDescriptorFactory_CouldNotFindMatchingEndBrace" xml:space="preserve">
+    <value>Could not find matching ']' for required attribute.</value>
```

Comment:
Point about referring to the attribute and its property applies below as well. "required attribute" isn't as meaningful.